Inferences/BackwardDemodulation.cpp

```diff
@@ -119,41 +126,27 @@ struct BackwardDemodulation::ResultFn
     }
 
     TermList lhs=arg.first;
+    TermList rhs=EqHelper::getOtherEqualitySide(_eqLit, lhs);
 
     // AYB there used to be a check here to ensure that the sorts
     // matched. This is no longer necessary, as sort matching / unification
     // is handled directly within the tree
 
-    TermList rhs=EqHelper::getOtherEqualitySide(_eqLit, lhs);
+    auto subs = qr.unifier;
+    ASS(subs->isIdentityOnResultWhenQueryBound());
```

Comment:
So this is always true now? Was it always so?

Reply:
Currently we only use `CodeTreeTIS` in `ForwardDemodulation` (resp. `SubstitutionTree` in `BackwardDemodulation`) for which this (respectively `isIdentityOnQueryWhenResultBound()`) holds. And `ResultSubstitution::isRenaming` also assumes this, so this makes the code more consistent IMO. We can put them back if the indices change (which I don't see happening soon).